core/app/models/spree/order.rb

```diff
@@ -668,6 +670,14 @@ def has_free_shipping?
 
     private
 
+    def reset_order_if_store_association_has_changed
```

Comment:
@damianlegawiec If an order has its associated store changed, should we empty the order of products?

**Example scenario**
Admin is on the phone to a customer and creates order while managing EU store, adds 3 products that are only scoped to the EU store.

Then the customer says he is in the US... the Admin then changes the orders store to the U.S Store.

In this scenario would it be best to empty the cart as in the code below and begin adding the items that are appropriate to the store?

Reply:
@MatthewKennedy hmm I think we should

1. Prevent moving complete Orders between Stores
2. Incomplete - empty the Order from Products not available in the target store

Reply:
@damianlegawiec

- Complete orders are not store switchable, as requested.
- I'll look into removing just the products that are not scoped to the new store on change.

Reply:
@MatthewKennedy let's have that out of scope of this PR - this can be handled independently :)

Reply:
@damianlegawiec 
Yeah sure,I think this would be best tackled in its own PR.

 I have it working as far as removing line items on store change if the line items are not available in the new store the order belongs to, but just worried about changing an orders line items on switching stores, and what that means in terms of adjustments, promotions, shipping.

Like you say, a dedicated PR for this might be best, especially for writing lots of tests for that.